aristanetworks/ansible-avd
# Doc(eos_cli_config_gen): Fix new_key paths for router_bgp to raise errors in case of a key conflict

## Change Summary

Fix `new_key` paths under `router_bgp` to make sure errors are raised in case both old and new data models are used

## Related Issue(s)

Fixes #4596 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
- Update `router_bgp` fragment of the `eos_cli_config_gen` schema to use relative paths
- Update `eos_cli_gonfig_gen_negative_unit_tests` scenario to include conflicting combinations for `router_bgp` fragment of the schema

## How to test
- Checkout latest `devel` branch
- Install deps
- `cd` into `ansible_collections/arista/avd` and run `molecule converge -s eos_cli_config_gen_negative_unit_tests -- --limit conflicting-deprecated-and-new-keys`
- Observe 10 x `DEPRECATION WARNING` although there are 10 collisions under `router_bgp`. Execution is not blocked.
- Checkout PR
- Re-install PYAVD 
- Re-run `molecule converge -s eos_cli_config_gen_negative_unit_tests -- --limit conflicting-deprecated-and-new-keys` and confirm 10 new errors raised due to the conflicting data model keys.

## Checklist

### User Checklist

- N/A

### Repository Checklist



- [x ] My code has been rebased from devel before I start
- [ x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)
